skiko/src/jvmMain/kotlin/org/jetbrains/skiko/RendezvousBroadcastChannel.kt

```diff
@@ -1,30 +1,45 @@
 package org.jetbrains.skiko
 
-import kotlinx.coroutines.CompletableDeferred
 import kotlinx.coroutines.channels.Channel
-import java.util.concurrent.atomic.AtomicReference
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.resume
 
 /**
  * Behaves as Channel<Unit>(Channel.RENDEZVOUS), but with ability to send value to all current consumers
- * (which await on `receive` method)
+ * (which await on `receive` method).
  */
 internal class RendezvousBroadcastChannel<T> {
     private val onRequest = Channel<Unit>(Channel.CONFLATED)
-    private val onResult = AtomicReference(CompletableDeferred<T>())
+    private val receivers = mutableListOf<Continuation<T>>()
+    private val receiversCopy = mutableListOf<Continuation<T>>()
```

Comment:
Why make it a field, not a local? It would mean two threads in send all will race on copy.